packages/view-types/heatmap/src/HeatmapSubscriber.js

```diff
@@ -240,7 +244,7 @@ export function HeatmapSubscriber(props) {
         useDevicePixels
         onHeatmapClick={onHeatmapClick}
       />
-      {!disableTooltip && (
+      {!disableTooltip && tooltipsVisible && (
```

Comment:
these is a corner case here, which is that if the user sets in the view config `disableTooltip` to `true`, `tooltipsVisible` will still be equal to true and the checkbox will be checked, even though in reality no tooltips will appear. Ultimately, if `disableTooltip` is set to `true`, it will take precedence over the checkbox and it may seem to the user that the checkbox is broken, because it will never show the tooltips. I can think of two ways to solve this:
1. Remove `disableTooltip` prop and only allow changing tooltip visibility settings from the menus of the plots. Potential problem here is that all view configs that set this prop will stop working. And users will need to set this every time they load their dataset, which could be annoying.
2. Make the initial default value of `tooltipsVisible` equal to `!disableTooltip`. This solves the problems above, looks like a better solution to me. I don't know if it is possible to do though. The fact that `disableTooltip` value can be different for each view type, but `tooltipsVisible` is shared will be another cause for edge cases.

What do you think @keller-mark?

Reply:
another thing we can do is not display the menu option to enable or disable tooltips in the menu of `disableTooltip` is true.

Reply:
I implemented this in commit 0b9a03b, what do you think?

Reply:
Thanks for identifying this corner case and proposing these solutions. 

I would go with option 1 (Remove disableTooltip prop) to simplify the logic that we would need to maintain and to eliminate confusion about which value (disableTooltip vs tooltipsVisible) would take priority.

As for the two concerns:

1. Potential problem here is that all view configs that set this prop will stop working.

This is true, but we have a bit of flexibility here since we do not actually document `props` anywhere (besides in the JSDoc comments), and this is deliberate. For example, in this discussion https://github.com/vitessce/vitessce/discussions/1232#discussioncomment-2793775 I noted
> Since we have several major planned features on our roadmap I wouldn't want to commit to not making any breaking changes to internal props.

Once we move something from `props` to coordinationSpace, we are effectively committing to longer-term support of it (and all [coordination types](http://vitessce.io/docs/coordination-types/) are automatically documented). This means we can use `props` as a bit of a test-bed / way to quickly implement something, for example if something needs to be rushed for the HuBMAP data portal. 

In the case that we want to avoid a breaking change, we can use view config versioning and [upgrade functions](https://github.com/vitessce/vitessce/blob/3e61ebdbde78291802b9e2a196182c9d4ca3aebb/packages/schemas/src/previous-config-upgraders.ts#L105) to convert from props to coordinationSpace without breaking previous view configs.

It is a bit of a case-by-case decision, but the main thing to consider is whether the `disableTooltip` prop is currently used for "production" use cases anywhere. For something like the Heatmap `transpose` prop, it is used, so an upgrade function would be more necessary. A good way to check this is to do a quick search in portal-visualization, which defines Vitessce configurations for the HuBMAP data portal:

- [search for "transpose"](https://github.com/search?q=repo%3Ahubmapconsortium%2Fportal-visualization%20transpose&type=code)
- [search for "disableTooltip"](https://github.com/search?q=repo%3Ahubmapconsortium%2Fportal-visualization+disableTooltip&type=code)
- [search for "disableChannelsIfRgbDetected"](https://github.com/search?q=repo%3Ahubmapconsortium%2Fportal-visualization%20disableChannelsIfRgbDetected&type=code)

In this case, `disableTooltip` is not used in any of those configurations, so it is relatively safe to remove the prop.
To be 100% sure we can always go the conservative route and define the upgrade function but I do not think it is necessary for `disableTooltip`.


2. And users will need to set this every time they load their dataset, which could be annoying.

I am not sure I understand this point, as the coordination space values can be initialized in the view config using the top-level `coordinationSpace` and view-level `coordinationScopes` properties.

Reply:
> > As for the two concerns:
> 
> Potential problem here is that all view configs that set this prop will stop working.
> This is true, but we have a bit of flexibility here since we do not actually document props anywhere (besides in the JSDoc comments), and this is deliberate. For example, in this discussion https://github.com/vitessce/vitessce/discussions/1232#discussioncomment-2793775 I noted
> 
> Since we have several major planned features on our roadmap I wouldn't want to commit to not making any breaking changes to internal props.
> 
> Once we move something from props to coordinationSpace, we are effectively committing to longer-term support of it (and all [coordination types](http://vitessce.io/docs/coordination-types/) are automatically documented). This means we can use props as a bit of a test-bed / way to quickly implement something, for example if something needs to be rushed for the HuBMAP data portal.

This makes a lot of sense now. I didn't realise that the purpose of `props` is mainly experimental. In this case, also considering the fact that nobody is using `disableTooltip` as props I agree that indeed the best option is to remove it. I have removed it in commit 90bfa7b.

> It is a bit of a case-by-case decision, but the main thing to consider is whether the disableTooltip prop is currently used for "production" use cases anywhere. For something like the Heatmap transpose prop, it is used, so an upgrade function would be more necessary. A good way to check this is to do a quick search in portal-visualization, which defines Vitessce configurations for the HuBMAP data portal:
> 
> [search for "transpose"](https://github.com/search?q=repo%3Ahubmapconsortium%2Fportal-visualization%20transpose&type=code)
> [search for "disableTooltip"](https://github.com/search?q=repo%3Ahubmapconsortium%2Fportal-visualization+disableTooltip&type=code)
> [search for "disableChannelsIfRgbDetected"](https://github.com/search?q=repo%3Ahubmapconsortium%2Fportal-visualization%20disableChannelsIfRgbDetected&type=code)

Thanks for the tip on search!

> 2. And users will need to set this every time they load their dataset, which could be annoying.
> I am not sure I understand this point, as the coordination space values can be initialized in the view config using the top-level coordinationSpace and view-level coordinationScopes properties.

Yeah, when I wrote this point I forgot that people can add coordinationScope properties in their view config. I imagined that they'd need to disable the tooltip from the settings menu every time. You are right, this point doesn't make sense.